inference-engine/thirdparty/clDNN/src/impls/onednn/primitive_onednn_base.h

```diff
@@ -67,28 +97,713 @@ struct typed_primitive_onednn_impl : public typed_primitive_impl<PType> {
         return !zp.empty() && (reinterpret_cast<const int32_t&>(zp[0]) == drsv);
     }
 
+    static dnnl::post_ops try_optimize_post_ops(const typed_program_node<PType>& arg, dnnl::post_ops& p_ops,
+                                                const std::shared_ptr<dnnl::primitive_attr>& attr,
+                                                bool& optimization_is_completed) {
+        // Get current node id for creating of optimization map
+        auto node_id = arg.id();
+
+        // Create new dnnl::post_ops object which will be filled inside the optimization process
+        dnnl::post_ops optimized_p_ops;
+
+        // Add new post-op into optimized_p_ops structure
+        auto add_post_op = [&](onednn_post_op_type type, const dnnl::post_ops& cur_p_ops, dnnl::post_ops& new_p_ops, int idx) {
+            switch (type) {
+                case onednn_post_op_type::eltwise_act:
+                case onednn_post_op_type::eltwise_clip:
+                case onednn_post_op_type::eltwise_linear:
+                case onednn_post_op_type::eltwise_round:
+                {
+                    dnnl::algorithm alg;
+                    float scale, alpha, beta;
+                    cur_p_ops.get_params_eltwise(idx, scale, alg, alpha, beta);
+                    new_p_ops.append_eltwise(scale, alg, alpha, beta);
+                    break;
+                }
+
+                case onednn_post_op_type::binary_add:
+                case onednn_post_op_type::binary_mul:
+                case onednn_post_op_type::binary_max:
+                case onednn_post_op_type::binary_min:
+                {
+                    dnnl::algorithm alg;
+                    dnnl::memory::desc desc;
+                    cur_p_ops.get_params_binary(idx, alg, desc);
+                    new_p_ops.append_binary(alg, desc);
+                    break;
+                }
+
+                case onednn_post_op_type::scale:
```

Comment:
Is scale not supported? Then I think it would be better to throw exception.

Reply:
No, it's supported. I could combine this case with 3 "case onednn_post_op_type::optimized" cases but thought that this variant is worse from a point of readability and code's understanding. What do you think about?

Reply:
I see. In that case, this looks fine.